apps/studio/src/components/tableview/TableTable.vue

```diff
@@ -640,6 +640,11 @@ export default Vue.extend({
               ? [true, false]
               : undefined,
             allowEmpty: true,
+            preserveObject: column.dataType.startsWith('_'),
```

Comment:
This looks like a good solution!

This will only fix this for Postgres though, correct? How do other database types identify arrays?

Reply:
Isn't the value an array to begin with? If so, can't it preserve object by default for all arrays?

Reply:
Yup! Since postgres is the only database that support arrays, so this is safe IMO. However, mysql support json. So I'm gonna need to look at that to finish up the PR.

Reply:
@azmy60 I don't know about other databases, but BigQuery does support arrays so maybe test that

Reply:
@azmy60 Hmm, if the mutator is converting from an array, that's probably not correct. I think we have a `preserveComplex` flag that is supposed to help fix that?

Good catch on the nullable editor turning the value into a string -- that definitely needs fixing. If the value is an array, or a json object, it should just keep it that way.

For background, Tabulator has 2 ways to manipulate values:
- Mutators - changes the ACTUAL value when set
- Formatters - only changes how the values are DISPLAYED

So mutators should def keep them as complex types when appropriate. See `genericMutator`. This has a `preserveComplex` argument that can be set to true. See also `resolveDataMutator`

`niceString` is only used in a formatter, so don't worry about that one.

Reply:
Sorry, this ended up way more complex than I thought it would be :-D

Reply:
Sorry, the `preserveObject` flag is a parameter for the nullable editor. It's the fix for turning the value back into it's original type when changing some values from cells.

As of `preserveComplex` in `genericMutator`, I had switched that to [true](https://github.com/beekeeper-studio/beekeeper-studio/pull/1708/files#diff-c43a89ad7180e5dcda2d74247d1a2823e3399ffd622b3fd5eee1af3b3ebd7b87L12-R15).

Reply:
This all look super helpful for an issue we're seeing with import where arrays are causing DB insert errors.